chore: Added data-test attributes for testing

adam-stasiak opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Added test attribute for cypress testing.
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
Added data-test attributes for testing

mistercrunch commented:
I'd merge quickly if checks passed as this is likely to conflict with everything. Wondering why you didn't approach this with a set of smaller PRs to start with.

rusackas commented:
Is this a re-work of #11254? Just trying to figure out if this needs to be thoroughly reviewed again, or if it's the same stuff, split out. I'd still honestly rather just finish up the prior one, if it's all the same. Not sure if you or others have feelings on this.

rusackas commented on superset-frontend/package-lock.json:
```diff
@@ -21137,8 +21137,7 @@
               "version": "2.1.1",
               "resolved": "",
               "integrity": "sha1-w7M6te42DYbg5ijwRorn7yfWVN8=",
-              "dev": true,
```
FWIW, I'm not sure what others do, but I just don't bother committing `package-lock.json` if `package.json` hasn't changed. The thrashing of this file doesn't _usually_ seem to make any notable difference other than some occasional merge conflicts. :)

Commit:
fixed incorrect naming change

adam-stasiak commented:
> Is this a re-work of #11254? Just trying to figure out if this needs to be thoroughly reviewed again, or if it's the same stuff, split out. I'd still honestly rather just finish up the prior one, if it's all the same. Not sure if you or others have feelings on this.

I fixed review findings and splitted change related to attribute from #11134 into this PR and next I will submit test code using these attributes. It will be easier now when this will be merged - smaller changes then I think. Please for green light for this and merging to avoid conflicts - real blocker for me in pushing forward.

rusackas commented on superset-frontend/src/components/FaveStar.tsx:
```diff
@@ -46,7 +46,12 @@ export default class FaveStar extends React.PureComponent<FaveStarProps> {
-          <a href="#" onClick={this.onClick} className="fave-unfave-icon">
+          <a
+            href="#"
+            onClick={this.onClick.bind(this)}
```
this is a very strange place to see a `bind` - if this needs to happen, pease do it in the constructor.

nytai replied:
this looks like a bad rebase. I removed these binds in https://github.com/apache/incubator-superset/pull/11101

adam-stasiak replied:
yes - bad rebase

rusackas commented on superset-frontend/src/components/dataViewCommon/TableCollection.tsx:
```diff
@@ -272,6 +276,7 @@ export default function TableCollection({
             const rowId = row.original.id;
             return (
               <tr
+                data-test="table-row"
```
I'm a little less enthused about these table-row and table-cell ones, since they don't add any specificity to the test, but I guess they don't hurt, meaning we can remove/alter the class if needed. In this case, it just feels like a little bloat to the code. But... for anyone else reading this, the plan is to remove these using [this package](https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id).

nytai replied:
hmm, how would we automate testing production/staging releases? I don't think these are really adding much bloat in the grand scheme of things. Data test attributes should be unique though, so I'd suggest changing this to `table-row-${i}` where I is the row number, so it's easier to select the nth row

adam-stasiak replied:
I was thinking about this but finally decided to left this as table-row and logic related to choosing proper element move to cypress code using children() and eq() syntax. What do you think? which option is more valuable?

nytai replied:
🤷 this is probably fine for now. We can easily switch if needed

rusackas replied:
Interesting point (and thanks for the comments in general), @nytai - I agree this is fine for the purposes of this PR... let's have a separate chat/PR where we talk about these next steps.

rusackas commented on superset-frontend/src/dashboard/components/Header.jsx:
```diff
@@ -384,7 +387,7 @@ class Header extends React.PureComponent {
         <div className="button-container">
           {userCanSaveAs && (
-            <div className="button-container">
+            <div className="button-container" data-test="button-container">
```
```suggestion
            <div className="button-container" data-test="dashboard-edit-actions">
```
Maybe something a bit more explicit like so?

adam-stasiak replied:
Yes it seems to be more reasonable

adam-stasiak replied:
I applied this suggestion! thanks!

rusackas commented on superset-frontend/src/dashboard/components/PropertiesModal.jsx:
```diff
@@ -346,6 +347,15 @@ class PropertiesModal extends React.PureComponent {
               >
                 {saveLabel}
               </Button>
+              <Button
```
This looks like a rebase error or something, as there are now TWO cancel buttons here.

adam-stasiak replied:
yes I deleted this now

rusackas commented on superset-frontend/src/explore/components/controls/DatasourceControl.jsx:
```diff
@@ -210,6 +210,7 @@ class DatasourceControl extends React.PureComponent {
               className=""
               bsSize="sm"
               id="datasource_menu"
+              data-test="datasource-menu"
```
hooray dashed-naming-convention! 🎉

rusackas commented:
Aside from the aforementioned tweaks/questions/suggestions, this LGTM. Let's tie up those loose ends and merge this thing! 🚀

Commit:
review changes

rusackas reviewed (approved):
LGTM!

rusackas commented:
I'll merge this if/when the light turns green :D
